res/res_pjsip/config_transport.c

```diff
@@ -916,22 +921,25 @@ static int transport_apply(const struct ast_sorcery *sorcery, void *obj)
 
 			temp_state->state->tls.sockopt_params.options[sockopt_count].level = pj_SOL_TCP();
 			temp_state->state->tls.sockopt_params.options[sockopt_count].optname = TCP_KEEPIDLE;
-			temp_state->state->tls.sockopt_params.options[sockopt_count].optval = &transport->tcp_keepidle_time;
-			temp_state->state->tls.sockopt_params.options[sockopt_count].optlen = sizeof(transport->tcp_keepidle_time);
+			temp_state->state->tls.sockopt_params.options[sockopt_count].optval = &transport->tcp_keepalive_idle_time;
+			temp_state->state->tls.sockopt_params.options[sockopt_count].optlen = sizeof(transport->tcp_keepalive_idle_time);
 			sockopt_count++;
 
 			temp_state->state->tls.sockopt_params.options[sockopt_count].level = pj_SOL_TCP();
 			temp_state->state->tls.sockopt_params.options[sockopt_count].optname = TCP_KEEPINTVL;
-			temp_state->state->tls.sockopt_params.options[sockopt_count].optval = &transport->tcp_keepintvl_time;
-			temp_state->state->tls.sockopt_params.options[sockopt_count].optlen = sizeof(transport->tcp_keepintvl_time);
+			temp_state->state->tls.sockopt_params.options[sockopt_count].optval = &transport->tcp_keepalive_interval_time;
+			temp_state->state->tls.sockopt_params.options[sockopt_count].optlen = sizeof(transport->tcp_keepalive_interval_time);
 			sockopt_count++;
 
 			temp_state->state->tls.sockopt_params.options[sockopt_count].level = pj_SOL_TCP();
-			temp_state->state->tls.sockopt_params.options[sockopt_count].optname = TCP_KEEPCNT;
-			temp_state->state->tls.sockopt_params.options[sockopt_count].optval = &transport->tcp_keepcnt;
-			temp_state->state->tls.sockopt_params.options[sockopt_count].optlen = sizeof(transport->tcp_keepcnt);
+			temp_state->state->tls.sockopt_params.options[sockopt_count].optname = tcp_keepalive_probe_count;
+			temp_state->state->tls.sockopt_params.options[sockopt_count].optval = &transport->tcp_keepalive_probe_count;
+			temp_state->state->tls.sockopt_params.options[sockopt_count].optlen = sizeof(transport->tcp_keepalive_probe_count);
 			sockopt_count++;
 		}
+#else
+	ast_log(LOG_WARNING, "TCP keepalive settings are disabled because PJ_MAX_SOCKOPT_PARAMS is less than 5.\n");
```

Comment:
Specify the transport name in this. It should also be more user friendly. "TCP keepalive settings can not be set due to PJSIP being built without support for setting all options. Consider using bundled PJSIP instead to allow functionality to work."